sugondat-shim/src/cli.rs

```diff
@@ -153,64 +148,104 @@ pub mod query {
+            if let Ok(n) = input.parse::<u64>() {
+                Ok(BlockRef::Number(n))
+            } else {
+                Err(format!("parse error. see `--help`"))
+            }
         }
+    }
 
-        impl std::fmt::Display for BlockRef {
-            fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-                match *self {
-                    BlockRef::Best => write!(f, "best"),
-                    BlockRef::Number(n) => write!(f, "{}", n),
-                    BlockRef::Hash(h) => write!(f, "0x{}", hex::encode(&h[..])),
-                }
+    fn decode_hash(input: &str) -> Result<Option<[u8; 32]>, String> {
+        if let Some(s) = input.strip_prefix("0x") {
+            let bytes =
+                hex::decode(s).map_err(|_| "Invalid parameter: not hex encoded".to_owned())?;
+
+            let mut hash = [0u8; 32];
+            if bytes.len() != 32 {
+                return Err("Invalid parameter: hash not 32 bytes".to_owned());
             }
+
+            hash.copy_from_slice(&bytes[..]);
+            Ok(Some(hash))
+        } else {
+            Ok(None)
         }
+    }
 
-        impl std::str::FromStr for BlockRef {
-            type Err = String;
+    pub mod blob {
+        use clap::Args;
 
-            fn from_str(input: &str) -> Result<Self, Self::Err> {
-                if input == "best" {
-                    return Ok(BlockRef::Best);
-                }
+        use super::{BlockRef, SugondatRpcParams};
 
-                if let Some(s) = input.strip_prefix("0x") {
-                    let bytes = hex::decode(s)
-                        .map_err(|_| "Invalid parameter: not hex encoded".to_owned())?;
+        #[derive(Debug, Args)]
+        pub struct Params {
+            #[clap(flatten)]
+            pub rpc: SugondatRpcParams,
 
-                    let mut hash = [0u8; 32];
-                    if bytes.len() != 32 {
-                        return Err("Invalid parameter: hash not 32 bytes".to_owned());
-                    }
+            /// The block to query information about.
+            ///
+            /// Possible values: ["best", number, hash]
+            ///
+            /// "best" is the highest finalized block.
+            ///
+            /// Hashes must be 32 bytes, hex-encoded, and prefixed with "0x".
+            #[arg(value_name = "BLOCK_REF")]
+            pub block: BlockRef,
 
-                    hash.copy_from_slice(&bytes[..]);
-                    return Ok(BlockRef::Hash(hash));
-                }
+            /// The index of the extrinsic (transaction) containing the blob.
+            #[arg(value_name = "INDEX")]
+            pub index: u32,
 
-                if let Ok(n) = input.parse::<u64>() {
-                    Ok(BlockRef::Number(n))
-                } else {
-                    Err(format!("parse error. see `--help`"))
-                }
-            }
+            /// Output the blob data as binary to stdout and nothing else.
```

Comment:
What that "else" could be? TBH not sure what you are trying to convey here.

Reply:
"else" is all the other stuff that gets printed as part of the output